helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -593,37 +610,32 @@ private void delayedRebalanceSchedule(ResourceControllerDataProvider clusterData
    * @param clusterData the cluster data cache.
    * @param resourceMap the rebalanaced resource map.
    * @param clusterChanges the detected cluster changes that triggeres the rebalance.
-   * @param resourceStatePriorityMap the state priority map for each resource.
    * @param baseline the baseline assignment
    */
   private void applyRebalanceOverwrite(Map<String, IdealState> idealStateMap,
       ResourceControllerDataProvider clusterData, Map<String, Resource> resourceMap,
       Map<HelixConstants.ChangeType, Set<String>> clusterChanges,
-      Map<String, Map<String, Integer>> resourceStatePriorityMap,
       Map<String, ResourceAssignment> baseline) throws HelixRebalanceException {
     Set<String> enabledLiveInstances = clusterData.getEnabledLiveInstances();
     // Note that the calculation used the baseline as the input only. This is for minimizing
     // unnecessary partition movement.
-    Map<String, ResourceAssignment> activeAssignment = calculateAssignment(clusterData,
-        clusterChanges, resourceMap, enabledLiveInstances, Collections.emptyMap(), baseline);
+    Map<String, IdealState> activeIdealStates = convertResourceAssignment(clusterData,
+        calculateAssignment(clusterData, clusterChanges, resourceMap, enabledLiveInstances,
```

Comment:
You mean applyRebalanceOverwrite? This method is not long I think.
If you mean the class itself, let's do it later. I think it would be a good idea. But not at this moment.